### ISSUE TITLE
timeout following chain led to needing to sync

### ISSUE THREAD

**Issue by rolfyone** [Consensys-Incorporated/teku#8198]
Timeout led to needing to drop back into sync mode, and took 10 minutes to recover the chain.
Is there anything else we should be doing here like retrying?

This was on discord but a fairly nice case with a log, so it's worth verifying the scenario, and brainstorming maybe.

It does note retry capacity exceeded, maybe its related to that?
```
Apr 13 07:57:39 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:57:39.001 INFO  - Slot Event  *** Slot: 8847286, Block: 7d5e3938267e9b2123edb5f153c1bf1c9284fd86685e6fb8a3842dac7dedc9e5, Justified: 276476, Finalized: 276475, Peers: 100
Apr 13 07:57:51 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:57:51.095 INFO  - Slot Event  *** Slot: 8847287, Block:                                                        ... empty, Justified: 276476, Finalized: 276475, Peers: 101
Apr 13 07:57:52 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:57:52.241 WARN  - blobSidecars validation result: result NOT_AVAILABLE, blob sidecars: 0, cause: Optional[TimeoutException]
Apr 13 07:57:52 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:57:52.241 WARN  - Unable to import block a3414d2f61e3967eff97e1ca9a74eaefda2b0af95b2f16151dbc72deafed2f81 (8847287) due to data unavailability
Apr 13 07:57:52 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:57:52.241 INFO  - Discarding block a3414d2f61e3967eff97e1ca9a74eaefda2b0af95b2f16151dbc72deafed2f81 (8847287) as data unavailable retry pool capacity exceeded
Apr 13 07:57:52 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:57:52.242 WARN  - Late Block Import *** Block: a3414d2f61e3967eff97e1ca9a74eaefda2b0af95b2f16151dbc72deafed2f81 (8847287) proposer 113214 arrival 1236ms, gossip_validation +2ms, pre-state_retrieved +3ms, processed +127ms, execution_payload_result_received +11ms, begin_importing +3862ms, completed +0ms
Apr 13 07:58:03 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:58:03.075 INFO  - Slot Event  *** Slot: 8847288, Block:                                                        ... empty, Justified: 276476, Finalized: 276475, Peers: 100
Apr 13 07:58:15 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:58:15.044 INFO  - Slot Event  *** Slot: 8847289, Block:                                                        ... empty, Justified: 276476, Finalized: 276475, Peers: 100
Apr 13 07:58:25 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:58:25.417 INFO  - Initiating pruning of finalized blocks prior to slot 7790432.
Apr 13 07:58:25 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:58:25.968 INFO  - Pruned 5000 finalized blocks prior to slot 7790432, last pruned slot was 7790431.
Apr 13 07:58:27 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:58:27.043 INFO  - Slot Event  *** Slot: 8847290, Block:                                                        ... empty, Justified: 276476, Finalized: 276475, Peers: 100
Apr 13 07:58:39 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:58:39.039 INFO  - Slot Event  *** Slot: 8847291, Block:                                                        ... empty, Justified: 276476, Finalized: 276475, Peers: 100
Apr 13 07:58:51 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:58:51.040 INFO  - Slot Event  *** Slot: 8847292, Block:                                                        ... empty, Justified: 276476, Finalized: 276475, Peers: 100
Apr 13 07:59:03 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:59:03.041 INFO  - Slot Event  *** Slot: 8847293, Block:                                                        ... empty, Justified: 276476, Finalized: 276475, Peers: 100
Apr 13 07:59:15 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:59:15.041 INFO  - Slot Event  *** Slot: 8847294, Block:                                                        ... empty, Justified: 276476, Finalized: 276475, Peers: 100
Apr 13 07:59:27 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:59:27.040 INFO  - Slot Event  *** Slot: 8847295, Block:                                                        ... empty, Justified: 276476, Finalized: 276475, Peers: 100
Apr 13 07:59:35 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:59:35.001 INFO  - Epoch Event *** Epoch: 276478, Justified checkpoint: 276477, Finalized checkpoint: 276476, Finalized root: bd26b985bf89ca7688e7931ed8ea34106765ea40587462311b75aa5ca4e9e7c3
Apr 13 07:59:35 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:59:35.001 INFO  - Sync committee performance: epoch 276477, expected 96, produced 0, correct 0, included 0 (0%)
Apr 13 07:59:36 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:59:36.281 INFO  - Attestation performance: epoch 276476, expected 500, produced 500, included 500 (100%), distance 1 / 1.02 / 2, correct target 500 (100%), correct head 500 (100%)
Apr 13 07:59:39 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:59:39.040 INFO  - Slot Event  *** Slot: 8847296, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 99
Apr 13 07:59:51 ovh-waw-2386G-1 teku[254451]: 2024-04-13 07:59:51.042 INFO  - Slot Event  *** Slot: 8847297, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:00:03 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:00:03.038 INFO  - Slot Event  *** Slot: 8847298, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:00:15 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:00:15.038 INFO  - Slot Event  *** Slot: 8847299, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:00:27 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:00:27.045 INFO  - Slot Event  *** Slot: 8847300, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:00:39 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:00:39.038 INFO  - Slot Event  *** Slot: 8847301, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:00:51 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:00:51.086 INFO  - Slot Event  *** Slot: 8847302, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 101
Apr 13 08:01:03 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:01:03.038 INFO  - Slot Event  *** Slot: 8847303, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:01:15 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:01:15.041 INFO  - Slot Event  *** Slot: 8847304, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:01:27 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:01:27.038 INFO  - Slot Event  *** Slot: 8847305, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:01:39 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:01:39.038 INFO  - Slot Event  *** Slot: 8847306, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:01:51 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:01:51.036 INFO  - Slot Event  *** Slot: 8847307, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:02:03 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:02:03.041 INFO  - Slot Event  *** Slot: 8847308, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 99
Apr 13 08:02:15 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:02:15.039 INFO  - Slot Event  *** Slot: 8847309, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:02:27 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:02:27.038 INFO  - Slot Event  *** Slot: 8847310, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:02:39 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:02:39.038 INFO  - Slot Event  *** Slot: 8847311, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:02:51 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:02:51.042 INFO  - Slot Event  *** Slot: 8847312, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:03:03 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:03:03.038 INFO  - Slot Event  *** Slot: 8847313, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:03:15 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:03:15.038 INFO  - Slot Event  *** Slot: 8847314, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 99
Apr 13 08:03:27 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:03:27.061 INFO  - Slot Event  *** Slot: 8847315, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:03:39 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:03:39.038 INFO  - Slot Event  *** Slot: 8847316, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:03:51 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:03:51.037 INFO  - Slot Event  *** Slot: 8847317, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:04:03 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:04:03.036 INFO  - Slot Event  *** Slot: 8847318, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:04:15 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:04:15.037 INFO  - Slot Event  *** Slot: 8847319, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 100
Apr 13 08:04:27 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:04:27.039 INFO  - Slot Event  *** Slot: 8847320, Block:                                                        ... empty, Justified: 276477, Finalized: 276476, Peers: 99
Apr 13 08:04:36 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:04:36.458 INFO  - Syncing started
Apr 13 08:04:36 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:04:36.536 INFO  - Syncing     *** Target slot: 8847321, Head slot: 8847286, Remaining slots: 35, Connected peers: 100
Apr 13 08:04:47 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:04:47.041 INFO  - Syncing     *** Target slot: 8847322, Head slot: 8847286, Remaining slots: 36, Connected peers: 98
Apr 13 08:04:59 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:04:59.037 INFO  - Syncing     *** Target slot: 8847323, Head slot: 8847286, Remaining slots: 37, Connected peers: 98
Apr 13 08:05:11 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:05:11.035 INFO  - Syncing     *** Target slot: 8847324, Head slot: 8847286, Remaining slots: 38, Connected peers: 99
Apr 13 08:05:23 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:05:23.037 INFO  - Syncing     *** Target slot: 8847325, Head slot: 8847286, Remaining slots: 39, Connected peers: 100
Apr 13 08:05:35 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:05:35.037 INFO  - Syncing     *** Target slot: 8847326, Head slot: 8847286, Remaining slots: 40, Connected peers: 100
Apr 13 08:05:47 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:05:47.110 INFO  - Syncing     *** Target slot: 8847327, Head slot: 8847286, Remaining slots: 41, Connected peers: 100
Apr 13 08:05:59 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:05:59.036 INFO  - Syncing     *** Target slot: 8847328, Head slot: 8847286, Remaining slots: 42, Connected peers: 100
Apr 13 08:05:59 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:05:59.038 INFO  - Attestation performance: epoch 276477, expected 500, produced 500, included 341 (68%), distance 1 / 1.00 / 1, correct target 341 (68%), correct head 341 (68%)
Apr 13 08:05:59 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:05:59.038 INFO  - Sync committee performance: epoch 276478, expected 96, produced 0, correct 0, included 0 (0%)
Apr 13 08:06:11 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:06:11.040 INFO  - Syncing     *** Target slot: 8847329, Head slot: 8847286, Remaining slots: 43, Connected peers: 99
Apr 13 08:06:23 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:06:23.037 INFO  - Syncing     *** Target slot: 8847330, Head slot: 8847286, Remaining slots: 44, Connected peers: 99
Apr 13 08:06:35 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:06:35.036 INFO  - Syncing     *** Target slot: 8847331, Head slot: 8847286, Remaining slots: 45, Connected peers: 97
Apr 13 08:06:43 ovh-waw-2386G-1 teku[254451]: 2024-04-13 08:06:43.894 WARN  - Encountered exception while processing message for topic /eth2/6a95a1a9/beacon_attestation_9/ssz_snappy
```

**Comment by tbenr** [comment]
This should be possible behaviour of version `24.3.0` and we fixed this in `24.3.1`. Waiting for user confirmation.

TLDR;
we was not correctly handling the retry pool for blocks erroring with `result NOT_AVAILABLE`. We were not pruning and not allowing new blocks pushing out old ones, so after filling it (10 blocks) the retry was essentially deactivated.

`24.3.1` fixes this. Moreover, next release will remove the pool retry completely.

**Comment by kernelpanic9** [comment]
That's correct. The version we used was 24.3.0.
Will upgrade. Thanks for the follow up!

**Comment by tbenr** [comment]
Closing for now. We will reopen it if you experience a similar behaviour with latest version.
